File: ios/Earwyrm/Core/Models/Follow.swift
import Foundation

struct Follow: Codable, Identifiable {
    let id: UUID
    let userId: UUID
    let filterType: String
    let filterValue: String
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case filterType = "filter_type"
        case filterValue = "filter_value"
        case createdAt = "created_at"
    }
}

struct FollowInsert: Encodable {
    let userId: UUID
    let filterType: String
    let filterValue: String

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case filterType = "filter_type"
        case filterValue = "filter_value"
    }
}

File: android/app/src/main/java/com/earwyrm/app/core/model/Follow.kt
package com.earwyrm.app.core.model

import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Follow(val id: String, @SerialName("user_id") val userId: String, @SerialName("filter_type") val filterType: String, @SerialName("filter_value") val filterValue: String, @SerialName("created_at") val createdAt: Instant)
@Serializable
data class FollowInsert(@SerialName("user_id") val userId: String, @SerialName("filter_type") val filterType: String, @SerialName("filter_value") val filterValue: String)
@Serializable
data class UserFollow(val id: String, @SerialName("follower_id") val followerId: String, @SerialName("following_id") val followingId: String, @SerialName("created_at") val createdAt: Instant)
@Serializable
data class UserFollowInsert(@SerialName("follower_id") val followerId: String, @SerialName("following_id") val followingId: String)
